src/fetchers/top-languages-fetcher.js

```diff
@@ -6,10 +6,10 @@ const fetcher = (variables, token) => {
   return request(
     {
       query: `
-      query userInfo($login: String!) {
+      query userInfo($login: String!, $ownerAffiliations: [RepositoryAffiliation]) {
         user(login: $login) {
-          # fetch only owner repos & not forks
-          repositories(ownerAffiliations: OWNER, isFork: false, first: 100) {
+          # do not fetch forks
+          repositories(ownerAffiliations: $ownerAffiliations, isFork: false, first: 100) {
```

Comment:
The reason for not using it is that ...... I didn't know this query type existed! I think this would be a more appropriate query, let me see what I can do.

Reply:
Ah, but there's a problem: if I occasionally contribute a bit of code to some repository that I don't maintain myself, doing so can lead to rather inaccurate results.

Reply:
That's correct, but you can exclude these repos manually as I said it is not 100% correct but closer to what you want. I saw a few people talking about a whitelist option both (whitelist or exclude list) are valid depending on how many repos you want to add to the each of these lists (do you have many repos to exclude or many to include)

Reply:
IMO, many users contribute small patches to open source projects they are interested in from time to time, and that doesn't mean they want the language distribution/number of stars of other codebases to be credited to their user stats.It would be too much of a hassle to add an exclusion repository every time a PR is submitted for a new codebase (as I'm doing now for this stats repo). For users with many personal repositories (I have many different private repositories), using a whitelist is also not a very convenient approach and can expose sensitive information in URL. I think using ownership-based fetch should be a more accurate way in this scenario.  What do you think?

Translated with www.DeepL.com/Translator (free version)

Reply:
I believe you have a valid point. It is a problem with no correct answer, but different type of answers based on the user.  This really comes down to different type of coders. 
1- Personal coder (lots of personal and private repos including private personal orgs)
2- Open Source coder (lots of public organization, personal and private repos including private personal  orgs)
3- Closed Source coder (lots of private organization, lots of public organizations, personal and private repos)

Each of these coders has a bit different requirements. So really to achieve the proper percentages we could define a route that each of these coders would be able to use and let the calculation to be dynamic based on that.

Reply:
I have developStorm's current version. It is simple and easy to understand.